content/razor-views/reading/creating-templates/index.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Creating a Template"
+date: 2023-02-09T12:48:24-06:00
+draft: false
+weight: 2
+originalAuthor: Courtney Frey # to be set by page creator
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+Using templates is a useful way to reduce the effort required to create and maintain a web-based project. Before you can dive into using templates, however, you need to take care of a little groundwork first.
+
+## Razor
+
+**Razor** is a templating syntax included in the application framework for our MVC project. It allows us to write C# code directly into an HTML tree.
+
+More information on Razor can be found in this [reference page](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/razor?view=aspnetcore-6.0).
+
+For the most part, Razor templates look and operate just like regular HTML code. Any C# logic we add to a template is preceded with the _at sign_, `@`. You can open any template in a browser and view it just like a static HTML file. Before we begin adding C# logic to Razor templates, we’ll first demonstrate creating a view using only HTML.
+
+In this chapter, you will construct some small practice projects to help you learn how to implement Razor templates.
+
+## Hello Views
+
+Before you start coding, we need to refactor `HelloController`. Now `Index()` is responding to `GET` requests at `localhost:5001/hello` and `Welcome()` is only responding to `POST` requests at `localhost:5001/hello`. Check out the code block below to refactor your code.
+
+Open up your `HelloASPDotNET` project in Visual Studio and make sure you have committed any recent changes. Make a new branch for creating views and code along with the reading.
+
```

Comment:
I liked what you did on the Using a template page with the drop down check your code.  Think that would work well here as well with the three code blocks.  Leave it up to you.

Reply:
I pulled them out of the drop down menu and instead created boxes for them to check their code.  i chose this option because some of 15 and most of 17 won't make sense using drop down menus.  feel that this provides more visual consistency.